### DISCUSSION TITLE
ES8388 setInputVolume() Calculation

### DISCUSSION THREAD

**Discussion by ki6syd** [pschatzmann/arduino-audio-driver#22] [Q&A] [answered]
I was wondering if you might be able to explain the [logic in AudioDriverES8388Class for setInputVolume()](https://github.com/pschatzmann/arduino-audio-driver/blob/aea146452cf9f53a6f3fffb341f094af33bceda9/src/Driver.h#L819), or maybe there is a bug?

It looks like `setInputVolume()` with an input integer from 0 to 100 results in a call to `es8388_set_mic_gain()` with `es_mic_gain_t` values that don't use the full 0 to +24dB analog gain, since the enum definitions have assignments in units of dB. Is there a reason that the ES8388 class divides by 10 rather than mapping 0-100% to 0-24_dB and then casting as `es_mic_gain_t` scaling?

Thanks!

Max

**Comment by pschatzmann** [comment, marked as answer]
Confirmed, it is a bug: I committed a [correction](https://github.com/pschatzmann/arduino-audio-driver/commit/488c16b20b0ccc4a80ce971ae4e78296e380194a).

**Reply by ki6syd** [reply]
Thanks for addressing that quickly!

**Reply by ki6syd** [reply]
@pschatzmann, reviving an old thread here...wanted to give a heads-up that I think there is an issue in line 825 of the file you linked above.

It is currently:

`int idx = limitValue(volume / 10, 9);`

But I believe it should be as follows:

`int idx = limitValue(volume / 10, 0, 9);`

**Comment by pschatzmann** [comment]
You are right - I committed a correction: the next time I suggest that you submit a pull request...

**Reply by ki6syd** [reply]
Sounds good, will do!
